tests/unit/customizations/s3/test_utils.py

```diff
@@ -228,6 +240,20 @@ def test_tell_and_seek(self):
         self.assertEqual(chunk.tell(), 0)
 
 
+class TestGuessContentType(unittest.TestCase):
+    def test_guess_content_type(self):
+        self.assertEqual(guess_content_type('foo.txt'), 'text/plain')
+
+    def test_guess_content_type_with_no_valid_matches(self):
+        self.assertEqual(guess_content_type('no-extension'), None)
+
+    def test_guess_content_type_with_unicode_error_returns_no_match(self):
+        with mock.patch('mimetypes.guess_type') as guess_type_patch:
```

Comment:
I really do not like when I have to patch stuff like this (as well in os.makedirs), but I am not sure if there is any better way to do this. I am up for suggestions.